### PR TITLE
ref(interfaces): Consolidate crumbs and images loaded search + filter

### PR DESCRIPTION
**Pull request by priscilawebdev**
Created a reusable search + filter component, now used by  the  breadcrumb and images loaded interfaces

### CODE REVIEW

**Comment by markstory** [comment]
One small UI issue I saw was the top border of the dropdown and the caret have an extra border.

Screen Shot 2021-06-30 at 3 48 49 PM

File: static/app/components/events/interfaces/breadcrumbs/utils.tsx [hunk 1]
```diff
@@ -13,7 +13,61 @@ import {
...
+
+function convertCrumbType(breadcrumb: Breadcrumb): Breadcrumb {
+  if (breadcrumb.type === BreadcrumbType.EXCEPTION) {
+    return {
+      ...breadcrumb,
+      type: BreadcrumbType.ERROR,
+    };
+  }
+  // special case for 'ui.' and `sentry.` category breadcrumbs
+  // TODO: find a better way to customize UI around non-schema data
```
Comment (markstory) [inline static/app/components/events/interfaces/breadcrumbs/utils.tsx:27]: Could you have a map of categories, and a fallback? It looks like the `sentry.transaction` and `sentry.event` keys could also be categories in the mapping.

**Review by markstory — COMMENTED** [review]
Nice job consolidating the two filters to share components :clap:

**Comment by priscilawebdev** [comment]
@markstory I tried to style it to be similar to the figma design, but you're right, the caret and the dropdown should not have borders between them. I'll update the code

**Comment by priscilawebdev** [comment]
it should be fine now :)
image

File: static/app/components/events/interfaces/breadcrumbs/utils.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/components/events/interfaces/breadcrumbs/utils.tsx:27]: hm, I'll think about it a little more and maybe update this logic in a follow-up PR ...  in this PR this code was just moved... I did not change anything

File: static/app/components/events/interfaces/breadcrumbs/utils.tsx [hunk 2]
```diff
@@ -13,7 +13,61 @@ import {
   IconWarning,
 } from 'app/icons';
 import {t} from 'app/locale';
-import {BreadcrumbType} from 'app/types/breadcrumbs';
+import {Breadcrumb, BreadcrumbLevelType, BreadcrumbType} from 'app/types/breadcrumbs';
```
Comment (priscilawebdev) [inline static/app/components/events/interfaces/breadcrumbs/utils.tsx:16]: functions have just been moved to this file and nothing has changed

**Review by matejminar — COMMENTED** [review]
image

These z-indexes might need tweaking.

**Comment by priscilawebdev** [comment]
thanks @matejminar it should be fine now
image

**Review by matejminar — APPROVED** [review]
Code-wise looks good to me 👍

**Pull request merged**